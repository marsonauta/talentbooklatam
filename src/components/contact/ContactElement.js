import React from 'react'

class ContactElement extends React.Component {

    constructor(props) {
        super(props)

        this.classes = new Map([['cv', 'contact-cv'], ['in', 'contact-in'], ['md', 'contact-wl'], [undefined, 'default']])
        this.images = new Map([['cv', 'fas fa-passport'], ['in', 'fab fa-linkedin'], ['md', 'fab fa-medium'], [undefined, 'default']])
        this.values = new Map([['cv', ' Curr√≠culum'], ['in', ' LinkedIn'], ['md', ' Medium'], [undefined, 'default']])

        this.state = {
            class: this.classes.get(this.props.type),
            image: this.images.get(this.props.type),
            value: this.values.get(this.props.type),
        }
    }

    render() {
        return <li className={this.state.class}>
            <a href={this.props.href} target="_blank">
                <i className={this.state.image} />
                {this.state.value}
            </a>
        </li>
    }
}

export default ContactElement