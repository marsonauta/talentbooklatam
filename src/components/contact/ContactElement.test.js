import React from 'react';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ContactElement from './ContactElement'

describe('ContactElement component', () => {

    test('contain hiperlink', () => {
        const expectedHref = "http://some.reference"

        const { container } = render(<ContactElement href={expectedHref} />)
        expect(container).toMatchSnapshot(`<a href="${expectedHref}"`)
    })

    describe('when the type is cv', () => {
        const type = 'cv'
        let container = undefined

        beforeEach(() => {
            container = render(<ContactElement type={type} />).container
        })

        test('the LI class contain contact-cv as value', () => {
            expect(container).toMatchSnapshot(`<li class="contact-cv">`)
        })

        test('the Image class contain fas fa-passport as value', () => {
            expect(container).toMatchSnapshot(`<i className="fas fa-passport"`)
        })
        test('the message is Curriculum', () => {
            expect(container).toMatchSnapshot(`/>Curr√≠culum</a>`)
        })
    })

})
