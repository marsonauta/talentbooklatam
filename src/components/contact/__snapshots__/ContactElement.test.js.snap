// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactElement component contain hiperlink: <a href="http://some.reference" 1`] = `
<div>
  <li
    class="default"
  >
    <a
      href="http://some.reference"
      target="_blank"
    >
      <i
        class="default"
      />
      default
    </a>
  </li>
</div>
`;

exports[`ContactElement component when the type is cv the Image class contain fas fa-passport as value: <i className="fas fa-passport" 1`] = `
<div>
  <li
    class="contact-cv"
  >
    <a
      target="_blank"
    >
      <i
        class="fas fa-passport"
      />
       Currículum
    </a>
  </li>
</div>
`;

exports[`ContactElement component when the type is cv the LI class contain contact-cv as value: <li class="contact-cv"> 1`] = `
<div>
  <li
    class="contact-cv"
  >
    <a
      target="_blank"
    >
      <i
        class="fas fa-passport"
      />
       Currículum
    </a>
  </li>
</div>
`;

exports[`ContactElement component when the type is cv the message is Curriculum: />Currículum</a> 1`] = `
<div>
  <li
    class="contact-cv"
  >
    <a
      target="_blank"
    >
      <i
        class="fas fa-passport"
      />
       Currículum
    </a>
  </li>
</div>
`;
